openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -267,13 +269,65 @@ def _does_name_change_require_verification(user, old_name, new_name):
     """
     If name change requires verification, do not update it through this API.
     """
+
+    user_profile = UserProfile.objects.get(user=user)
+    profile_meta = user_profile.get_meta()
+    name_changes = 0
+    if 'old_names' in profile_meta:
+        name_changes = len(profile_meta['old_names'])
+
     return (
         is_verified_name_enabled()
-        and old_name != new_name
-        and len(get_certificates_for_user(user.username)) > 0
+        and (
+            _edits_are_invalid(old_name, new_name)
+            or name_changes >= 2
+            or len(get_certificates_for_user(user.username)) > 0
+        )
     )
 
 
+def _edits_are_invalid(old_name, new_name):
```

Comment:
boolean nit: humans generally expect true to mean go ahead and false to mean you lose so _edits_are_allowed makes more sense reading quickly than edits_are_forbidden